github_helper.py

```diff
@@ -23,30 +22,26 @@ def __init__(self, org, repo):
         self.repo_url = GITHUB_REPO_FMT_URL.format(org, repo)
         with open(GITHUB_SECRET, 'r') as key_file:
             self.bot_key = key_file.read().strip()
-        _print_flush('Loaded key file.')
 
     def fetch_prs(self):
         """Gets the PRs for the configured repository.
 
         Returns:
-            An iterable containing instances of GithubPR if successful.
-            Empty list otherwise.
+            A tuple of (pr_list, error).
+            pr_list contains an iterable containing instances of GithubPR if
+            successful; empty list otherwise.
+            error is None if no errors, else error message.
         """
         try:
             prs = self.get(GITHUB_PULLS_ENDPOINT)
-            return [GithubPR(self, pr) for pr in prs]
+            return [GithubPR(self, pr) for pr in prs], None
         except HTTPError as exc:
-            _print_flush('Non-200 HTTP Code Received:')
-            _print_flush(exc)
+            return [], 'Non-200 HTTP Code Received: {}'.format(exc)
         except Timeout as exc:
-            _print_flush('Request timed out:')
-            _print_flush(exc)
+            return [], 'Request timed out: {}'.format(exc)
         except RequestException as exc:
-            _print_flush('Catastrophic error in requests:')
-            _print_flush(exc)
-        _print_flush('Retrieving pull requests failed.')
-        return []
-
+            return [], 'Catastrophic error in requests: {}'.format(exc)
+        return [], 'Unknown Error'
```

Comment:
I think agree, but here we aren't handling them in the helper code. We're returning an error (rather than raising it) and expecting the caller to handle it appropriately. Either (1) the caller should be able to ignore errors (in which case we log, and return `[]`) or (2) the caller should have to handle errors, in which case we may as well throw. The only reason to use this pattern is if we want to both return an error and return information, but we're not doing that.